src/sonic-config-engine/config_samples.py

```diff
@@ -9,6 +9,39 @@
 else:
     UNICODE_TYPE = unicode
 
+
+# The new enhancements to this file provides capabilities to generate l1 & l3
+# configuration in addition the existing l2, t1 and empty configurations
+# So the following capabilities exits now:
+#    't1': generate_t1_sample_config,
+#    'l2': generate_l2_config,
+#    'empty': generate_empty_config,
+#    'l1': generate_l1_config,
+#    'l3': generate_l3_config
+
+def generate_l1_config(data):
+    data['DEVICE_METADATA']['localhost']['hostname'] = 'sonic'
+    data['DEVICE_METADATA']['localhost']['type'] = 'LeafRouter'
+    for port in natsorted(data['PORT']):
+        data['PORT'][port]['admin_status'] = 'up'
+        data['PORT'][port]['mtu'] = '9100'
+    return data;
+
+def generate_l3_config(data):
+    data['DEVICE_METADATA']['localhost']['hostname'] = 'sonic'
+    data['DEVICE_METADATA']['localhost']['type'] = 'LeafRouter'
+    data['DEVICE_METADATA']['localhost']['bgp_asn'] = '65100'
+    data['LOOPBACK_INTERFACE'] = {"Loopback0": {},
+                                  "Loopback0|10.1.0.1/32": {}}
```

Comment:
can you remove those hardcoded 'type', 'bgp_asn' and 'LOOPBACK_INTERFACE'?

Reply:
Please review https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/config_samples.py 
The new methods for L1 & L3 are just extensions to the existing methods and has the minimum set of parameters required for L1 & L3 System bringup.

Reply:
Tracked thru #7637